asdf/_tests/test_lazy_nodes.py

```diff
@@ -408,3 +411,26 @@ def test_lazy_generator_converter(tmp_path, lazy_generator_class):
 
     with asdf.open(fn, lazy_tree=True) as af:
         assert isinstance(af["obj"].data, dict)
+
+
+def test_lazy_copy(tmp_path):
+    """
+    Test that copying an AsdfFile instance with a lazy
+    tree doesn't result in the copy retaining references
+    to the instance.
+    """
+    fn = tmp_path / "test.asdf"
+    obj = asdf.tags.core.IntegerType(1)
+    tree = {"a": {"b": obj}}
+    # make a recursive structure
+    tree["a"]["c"] = tree["a"]
+
+    asdf.AsdfFile(tree).write_to(fn)
+
+    with asdf.open(fn, lazy_tree=True) as af:
+        af2 = af.copy()
```

Comment:
should we also test multiple files open in context managers? To replicate the original issue (or was that using open files?)